### DISCUSSION TITLE
Wpf 动画如何由ViewModel里面的属性触发呢？

### DISCUSSION THREAD

**Discussion by yangf85** [BYJRK/DotNet-Discussions#54] [WPF 开发]
比如我有一个登录按钮，登录完成以后,viewmodel里面有个属性发生了改变，由这个属性来控制播放一段动画, 应该如何操作啊？
我能想到的是使用附加属性来做

**Comment by chaojicpu** [comment]
道友可以把你使用附加属性来做代码放上来学习一下吗？

之前做法，在ViewModel中控制Visibility来显示隐藏前台容器控件，类似弹窗那种

借道友贵地 求助：控件Visibility值变化时实现渐隐动画效果

万谢

**Comment by BYJRK** [comment]
这些都可以考虑写在 View 的 Style 或 Template 的 DataTrigger 当中，然后添加 EnterAction 与 ExitAction，并在里面写 Storyboard，或者用 Behaviors 中的 Trigger 也可以

**Reply by chaojicpu** [reply]
稽首，谢 真人 解惑
